### ISSUE TITLE
Implement Rating Battle

### ISSUE THREAD

**Issue by mm201** [mm201/pkmn-classic-framework#111]
Some research has taken place in #8. Apparently, most of Rating Battle takes place in the SAKE server which doesn't need much server logic and is well understood. The `worldbattle` endpoints seem to just deal with savefiles again?

**Comment by zurgeg** [comment]
> Some research has taken place in #8. Apparently, most of Rating Battle takes place in the SAKE server which doesn't need much server logic and is well understood. The `worldbattle` endpoints seem to just deal with savefiles again?

Nope. Worldbattle is DLS and SAKE

**Comment by mm201** [comment]
dls1 is also a well-understood and backed up server so that's fine.

**Comment by zurgeg** [comment]
> dls1 is also a well-understood and backed up server so that's fine.

yeah. I know. We also have the regcards figured out partially so we're good there

**Comment by Isaac0-dev** [comment]
I was able to get far enough into the rated battle connection process to see it searching for players. 
<img width="279" height="454" alt="Image" src="https://github.com/user-attachments/assets/e499a403-42d0-4550-b1c1-d55d4878b7cf" />


So I thought i'd try to document what I've learned about the connection process:

1. connection to `en-ds.pokemon-gl.com`, querying `worldbattle.download`
   While I don't understand this structure entirely, I know you have to respond with 8 bytes.
   I believe it to be like this:
   4 bytes of 0, No idea what this is.
   1 byte, set to 0, for some flag (the game checks if it's 1 somewhere).
   1 byte, set to 0, may have something to do with whether you're signed up for a tournament (I've never seen that system active)
   1 byte, set to 1, flag that represents whether the connection worked i believe. Fails if it's not 1.
   1 byte, set to 0. No idea what this is.

2. POST to https://pkvldtprod.nintendo.co.jp/pokemon/validate
   This is the Pokemon validity check, already "implemented" by this project.

3. connection to syachi2ds.secure.sake.gs.nintendowifi.net, with an action of "GetMyRecords".
   This is the SAKE server request as you guys observed. It's basically a SOAP xml request.
   It's really handy because it sends through a list of all the fields it wants.
   So you can literally just respond with dummy values as 0.
   Except for "recordid". If you repond with 0, it will follow that up with a "CreateRecord" request since you basically told it there's no records for that player. So set that to anything that isn't 0.
